lib/dns.js

```diff
@@ -377,28 +394,30 @@ function setServers(servers) {
   }
 }
 
-const defaultResolver = new Resolver();
+let defaultResolver;
+
+function getDefaultResolver() {
+  return defaultResolver;
+}
+
+function setDefaultResolver(resolver) {
+  if (!(resolver instanceof Resolver)) {
+    throw new errors.TypeError('ERR_INVALID_DNS_RESOLVER');
+  }
+
+  defaultResolver = resolver;
+  DEFAULT_RESOLVER_EXPORTED_FUNCTIONS.forEach((key) => {
+    module.exports[key] = defaultResolver[key].bind(defaultResolver);
+  });
```

Comment:
This feels... hackish. Perhaps changing the module.exports properties into getters that return the equivalent method from `defaultResolver` would be a cleaner approach?

e.g. 

```js
module.exports = {
  /** ... **/
  get getServers() {
    return defaultResolver.getServers.bind(defaultResolver);
  }
  /** ... **/
}
```

That way the module's exported properties never change.